fedimint-core/src/util/mod.rs

```diff
@@ -307,13 +309,67 @@ pub fn handle_version_hash_command(version_hash: &str) {
     }
 }
 
+/// Run the supplied closure `op_fn` up to `max_attempts` times. Wait for the
+/// supplied `Duration` `interval` between attempts
+///
+/// # Returns
+///
+/// - If the closure runs successfully, the result is immediately returned
+/// - If the closure did not run successfully for `max_attempts` times, the
+///   error of the closure is returned
+pub async fn retry<F, Fut, T>(
+    op_name: impl Into<String>,
+    op_fn: F,
+    interval: Duration,
+    max_attempts: u32,
```

Comment:
max_attempts < timeout , just saying.

Isn't this `poll` from `devimint` where I just added deadlines and turned interval to be growing? Can we move it here from there and rename?

Reply:
That one uses `ControlFlow`, while here we want that to be implicitly implied by `Result`. `retry` could be implemented using `poll` though.

Reply:
That control flow there is a bit of a PITA, BTW (because of type inference on two variants). Not sure how badly we need it. In testing if there's an error that should really terminate polling... maybe we should just panic, instead of complicating common case? Just thinking aloud.

But one way or the other - would prefer to see a timeout/deadline approach here.

Reply:
Will leave it to @maan2003, don't have a strong opinion here.

Reply:
:shrug: devimint and this retry can just stay separate, attempt + interval is pretty close to deadline though